### ISSUE TITLE
Introduce InterSystems IRIS driver

### ISSUE THREAD

**Issue by evshvarov** [dbeaver/dbeaver#6289]
Hi dbeaver developers!
you already have InterSystems Caché support.
But InterSystems have introduced a new data platform InterSystems IRIS which has similar ODBC/JDBC driver.
IRIS is faster and has more capabilities.
Could you please add the support for IRIS?

IRIS Community verison on docker container could be found [here](https://hub.docker.com/_/intersystems-iris-data-platform)
Any questions could be discussed on InterSystems developers [forum](https://community.intersystems.com)

**Comment by serge-rider** [comment]
It seems that JDBC driver is not available for public download (there is no Maven repo or some direct download link). Maybe you are aware of any community resource for that?

Of course we can just make users to download IRIS by themselves and then to configure driver manually but this is actually a blocker for many people :(

**Comment by eduard93** [comment]
Here's [the driver](https://github.com/intersystems/quickstarts-multimodel-java/blob/master/lib/intersystems-jdbc-3.0.0.jar)

The schema is:
```
                <driver
                    id="iris"
                    label="InterSystems IRIS"
                    icon="icons/cache_icon.png"
                    iconBig="icons/cache_icon_big.png"
                    class="com.intersystems.jdbc.IRISDriver"
                    sampleURL="jdbc:IRIS://{host}:{port}/{database}"
                    defaultPort="51773"
                    webURL="http://www.intersystems.com/try/"
                    description="InterSystems IRIS Data Platform"
                    categories="sql,object">
                    <file type="jar" path="repo:/drivers/cache/CacheDB.jar"/>
                    <file type="license" path="repo:/drivers/cache/LICENSE.txt"/>
                </driver>
```

**Comment by tom-dyar** [comment]
I am happy to make a pull request, if that is what is needed to get this in. If someone can list a concise set of changes that need to be made in addition to the xml snippet above, that would be very helpful. For example, where would we put the jar? I forked the repo, but that seems to have broken "maven package", which builds fine from the base repo. Do I need to develop in Eclipse? Thanks!

**Comment by serge-rider** [comment]
We'll try include it in the next version.
XML config is enough, thanks.

**Comment by tom-dyar** [comment]
That is great, thanks!!

**Comment by tom-dyar** [comment]
We do have a new jar file, and wil be posting to maven at some point. Please do not use the jar linked above!

**Comment by tom-dyar** [comment]
Also, we have some custom SQL syntax that we would like to have highlighting for in the editor -- where do we specify that, is that something #6528 will address, or is there a mechanism already?

**Comment by serge-rider** [comment]
@tom-dyar Ok, thanks for the information.
In order to provide extra SQL syntax or some additional metadata entities we'll need to create IRIS extension.
But this can be done in a separate ticket. For now we need just driver config.

**Comment by bdeboe** [comment]
Just to be sure we're not being late with anything: does "wait for response" mean that *we* ought to wait or that *you* are waiting on us (InterSystems) for something? :-)
If the latter, can you clarify what's missing from the config posted by @eduard93 above? Please let us know how we can help.

Thanks!
benjamin

**Comment by tom-dyar** [comment]
Hi Benjamin,

I connected BobK to DBeaver and we have to wait until the newer jar is available on maven, I believe.

@Robert Kuszewski<mailto:Robert.Kuszewski@intersystems.com> - any estimate about when that might happen, and perhaps we can just go ahead and hand-deliver the latest to DBeaver?...

Thomas Dyar
Product Specialist, Machine Learning | InterSystems
+1.919.599.8874
http://intersystems.com/try
________________________________

**Comment by guilhermeledes** [comment]
It seems to be ready: https://mvnrepository.com/artifact/com.intersystems/intersystems-jdbc/3.0.0

**Comment by tom-dyar** [comment]
That is the old one – Robert, any update on this, thought I might have seen something about clients being published externally....

**Comment by evshvarov** [comment]
Colleagues, we are working on the automatic distribution of jdbc drivers to
maven repo. For now please download the JDBC driver from here:
https://github.com/intersystems-community/iris-driver-distribution/blob/main/intersystems-jdbc-3.1.0.jar
Thank you!

**Comment by rberge** [comment]
When will this be available for PC users and/or what is the recommended set up in the interim?

**Comment by mattcargile** [comment]
Is [this](https://community.intersystems.com/post/dbeaver-officially-supports-intersystems-iris) supported now? Haven't installed the Community edition yet. The aforementioned link's versioning appears to mimic the EE version? There is a separate dev channel for EE an Community, right?

EDIT: I installed the software and do see IRIS in the the connection picker UI, FYI. Don't have an instance to test on though.

**Comment by rberge** [comment]
I should remove myself from this because I no longer use intersystems.

**Comment by serge-rider** [comment]
It was fixed in 22.3, DBeaver CE now has new Intersystems driver config (mentioned above).
Closing
